In [17]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [18]:
parlamentarios_info = pd.read_csv('participacion/parlamentarios_info_general.csv')
descripcion_2023_1 = pd.read_csv('participacion/participacion_descripcion_2023_1.csv')
descripcion_2023_2 = pd.read_csv('participacion/participacion_descripcion_2023_2.csv')
descripcion_2024 = pd.read_csv('participacion/participacion_descripcion_2024.csv')

In [19]:
combined_df = pd.concat([descripcion_2023_1, descripcion_2023_2, descripcion_2024], ignore_index=True)


total_rows = combined_df.shape[0]
unique_values = combined_df["DESCRIPCION_DEBATE"].nunique()
print(f"'DESCRIPCION_DEBATE' distintos: {unique_values} en {total_rows} filas")


'DESCRIPCION_DEBATE' distintos: 5932 en 68161 filas


In [20]:
top_5_topics = combined_df['DESCRIPCION_DEBATE'].value_counts().head(5)
print(top_5_topics)

DESCRIPCION_DEBATE
OTROS DOCUMENTOS DE LA CUENTA                                                                                                                                                                                                                                                                                                                                                                                                                  9598
PETICIONES DE OFICIO. ARTÍCULOS 9° Y 9° A DE LA LEY ORGÁNICA CONSTITUCIONAL DEL CONGRESO NACIONAL.                                                                                                                                                                                                                                                                                                                                             2028
PETICIONES DE OFICIOS. ARTÍCULOS 9° Y 9° A DE LA LEY ORGÁNICA CONSTITUCIONAL DEL CONGRESO NACIONAL.          

Hay que separar los 5932, ver si se va a hacer con keywords 

In [21]:
# Normalizar los temas en la columna 'DESCRIPCION_DEBATE'
combined_df['DESCRIPCION_DEBATE'] = combined_df['DESCRIPCION_DEBATE'].str.lower()  # Convertir a minúsculas
combined_df['DESCRIPCION_DEBATE'] = combined_df['DESCRIPCION_DEBATE'].str.strip()  # Eliminar espacios al inicio y al final
combined_df['DESCRIPCION_DEBATE'] = combined_df['DESCRIPCION_DEBATE'].str.replace(r'\s+', ' ', regex=True)  # Reemplazar múltiples espacios por uno

# Verificar los valores únicos después de la normalización
unique_values_normalized = combined_df["DESCRIPCION_DEBATE"].nunique()
print(f"'DESCRIPCION_DEBATE' distintos después de normalización: {unique_values_normalized}")

'DESCRIPCION_DEBATE' distintos después de normalización: 5921


Esto no sirve directamente, pero es para saber que hay varios que tienen detalles así y que al revisarlo por texto sale similiar